randr.c

```diff
@@ -21,15 +21,51 @@
  */
 
 #include "xplugd.h"
+#include "edid.h"
 
 static char *con_actions[] = { "connected", "disconnected", "unknown" };
 
+int get_monitor_name(char *name, char *monitor_name)
+{
+	char path[255];
+	FILE *fp = NULL;
+	size_t result;
+	unsigned char edid[128];
+	MonitorInfo *info;
+
+	snprintf(path, sizeof(path), "/sys/class/drm/card0-%s/edid", name);
+	syslog(LOG_DEBUG, "DRM device sysfs path %s\n", path);
+	fp = fopen(path, "rb");
+	if (!fp) {
+		syslog(LOG_ERR, "Failed to find sys path for %s\n", name);
+		return -1;
+	}
+
+	result = fread (edid, 1, sizeof(edid), fp);
+	if (result != 128) {
+		syslog(LOG_NOTICE, "No EDID data found");
+		syslog(LOG_DEBUG, "DRM device sysfs path %s\n", path);
+		return -1;
+	}
+
+	info = decode_edid(edid);
+	if (info == NULL) {
```

Comment:
You changed style here, above you used `if (!fp)`, not `if (fp == NULL)` ... consistency is always good. Point of note, I prefer the former: `if (!info)` in this case.